static/js/settings_org.js

```diff
@@ -418,106 +418,177 @@ function _set_up() {
         settings_ui.disable_sub_setting_onchange(this.checked, "id_realm_message_content_edit_limit_minutes", true);
     });
 
-    exports.save_organization_settings = function () {
-        _.each(property_types.settings, function (v, k) {
-            property_type_status_element(k).hide();
-        });
+    function property_value_element_refers(property_name) {
+        if (property_name === 'realm_message_content_edit_limit_minutes') {
+            return Math.ceil(page_params.realm_message_content_edit_limit_seconds / 60);
+        }
+        return;
+    }
 
-        var message_editing_status = $("#admin-realm-message-editing-status").expectOne();
-        // grab the first alert available and use it for the status.
-        var $alerts = $(".settings-section.show .alert").hide();
-        // grab the first alert available and use it for the status.
-        var status = $("#admin-realm-notifications-stream-status");
-
-        var compose_textarea_edit_limit_minutes = $("#id_realm_message_content_edit_limit_minutes").val();
-        var new_allow_message_editing = $("#id_realm_allow_message_editing").prop("checked");
-
-        // If allow_message_editing is unchecked, message_content_edit_limit_minutes
-        // is irrelevant.  Hence if allow_message_editing is unchecked, and
-        // message_content_edit_limit_minutes is poorly formed, we set the latter to
-        // a default value to prevent the server from returning an error.
-        if (!new_allow_message_editing) {
-            if ((parseInt(compose_textarea_edit_limit_minutes, 10).toString() !==
-                 compose_textarea_edit_limit_minutes) ||
-                compose_textarea_edit_limit_minutes < 0) {
-            // Realm.DEFAULT_MESSAGE_CONTENT_EDIT_LIMIT_SECONDS / 60
-            compose_textarea_edit_limit_minutes = 10;
+    exports.extract_property_name = function (elem) {
+        return elem.attr('id').replace("id_", "").split('-').join('_');
+    };
+
+    function check_property_changed(elem) {
+        elem = $(elem);
+        var property_name = exports.extract_property_name(elem);
+        var current_val = page_params[property_name];
+        var changed_val;
+        if (typeof current_val === 'boolean') {
+            changed_val = elem.prop('checked');
+        } else if (typeof current_val === 'string') {
+            changed_val = elem.val().trim();
+        } else if (typeof current_val === 'number') {
+            current_val = current_val.toString();
+            changed_val = elem.val().trim();
+        } else {
+            // Check whether the id refers to a property whose name we can't
+            // extract from element's id.
+            current_val = property_value_element_refers(property_name).toString();
+            changed_val = elem.val().trim();
```

Comment:
I think this code path will crash if `current_val` is undefined (and actually the `toString` will crash if `property_value_element_refers` is undefined); seems probably buggy.